frontend/test/metabase/scenarios/question/custom_column.cy.spec.js

```diff
@@ -692,4 +692,34 @@ describe("scenarios > question > custom columns", () => {
     cy.contains("37.65");
     cy.findByText("No discount");
   });
+
+  it.skip("should work with relative date filter applied to a custom column (metabase#16273)", () => {
+    cy.intercept("POST", "/api/dataset").as("dataset");
+
+    openOrdersTable({ mode: "notebook" });
+    cy.findByText("Custom column").click();
+    popover().within(() => {
```

Comment:
Would you agree that using line spaces would help the brain chunk out discrete parts of the test?

Reply:
It's probably a personal preference and it comes down to what people got used to.
I find it much harder to follow/read if everything is separated with a single line.

Also, I tend to separate some chunks that are related in larger repros.

Reply:
Maybe just as feedback from one consumer of e2e test code (please take it as 1 data point :)))) ):

I spent this entire week working off of the e2e tests, to guide myself and find elements in the implementation code.

1. It was miraculous to have the e2e tests, otherwise basically I'd never finish the ticket.
2. overall the e2e test code is very easy to follow, which helps immensely
3. in tests with many discrete steps, when we have to run it dozens and dozens of times, making small advances, line spaces help in visually block the user steps (yes, they can be extracted to functions too...also can be very helpful)
4. so when seeing 26 lines in a row without a space, imho it is expected to be hard for the reader to chunk down each step.

```js
      openOrdersTable({ mode: "notebook" });
    cy.findByText("Custom column").click();
    popover().within(() => {
      cy.get("[contenteditable='true']")
        .type("case([Discount] >0, [Created At], [Product → Created At])")
        .blur();
      cy.findByPlaceholderText("Something nice and descriptive").type(
        "MiscDate",
      );
      cy.button("Done").click();
    });
    cy.findByText("Filter").click();
    popover()
      .contains("MiscDate")
      .click();
    // The popover shows up with the default value selected - previous 30 days.
    // Since we don't have any orders in the Sample Dataset for that period, we have to change it to the previous 30 years.
    cy.findByText("Days").click();
    cy.findByText("Years").click();
    cy.button("Add filter").click();
    cy.button("Visualize").click();
    cy.wait("@dataset").then(interception => {
      expect(interception.response.body.error).not.to.exist;
    });
    cy.findByText("MiscDate");
  });
});
```

Reply:
Do comments help visually break down the test for you (in the absence of well named helper functions) or do you still prefer the spaces?
```javascript
    openOrdersTable({ mode: "notebook" });
    // Add the custom column
    cy.findByText("Custom column").click();
    popover().within(() => {
      cy.get("[contenteditable='true']")
        .type("case([Discount] >0, [Created At], [Product → Created At])")
        .blur();
      cy.findByPlaceholderText("Something nice and descriptive").type(
        "MiscDate",
      );
      cy.button("Done").click();
    });
    // Filter on the previously defined custom column
    cy.findByText("Filter").click();
    popover()
      .contains("MiscDate")
      .click();
    // The popover shows up with the default value selected - previous 30 days.
    // Since we don't have any orders in the Sample Dataset for that period, we have to change it to the previous 30 years.
    cy.findByText("Days").click();
    cy.findByText("Years").click();
    cy.button("Add filter").click();
    // Preview the results
    cy.button("Visualize").click();
    cy.wait("@dataset").then(interception => {
      expect(interception.response.body.error).not.to.exist;
    });
    cy.findByText("MiscDate");
```

Obviously, something like this is much preferable (and we'll have it as soon as `release-x.39.x` gets merged into `master`):
```javascript
    openOrdersTable({ mode: "notebook" });

    addCustomColumn({
      formula: "case([Discount] >0, [Created At], [Product → Created At])",
      name: "MiscDate",
    });

    // Filter on the previously defined custom column
    cy.findByText("Filter").click();
    popover()
      .contains("MiscDate")
      .click();
    // The popover shows up with the default value selected - previous 30 days.
    // Since we don't have any orders in the Sample Dataset for that period, we have to change it to the previous 30 years.
    cy.findByText("Days").click();
    cy.findByText("Years").click();
    cy.button("Add filter").click();

    cy.button("Visualize").click();
    cy.wait("@dataset").then(interception => {
      expect(interception.response.body.error).not.to.exist;
    });
    cy.findByText("MiscDate");
```

Reply:
For basic code organization, spaces.